src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -45,14 +49,89 @@ static Startup()
         {
             Triggers<BuildChannel>.Inserted += entry =>
             {
-                BuildChannel entity = entry.Entity;
                 DbContext context = entry.Context;
+                BuildChannel entity = entry.Entity;
+
+                StartAssociatedReleasePipelinesAsync(entry).GetAwaiter().GetResult();
```

Comment:
We'll make this method a new service. The service will be called in a similar way as we do for DependencyUpdater below.  Tracked by this issue: https://github.com/dotnet/arcade/issues/1994

Reply:
GetAwaiter().GetResult() is a recipe for deadlocking the service.  Is this really something that can't be synchronous?

Reply:
Especially given there are web calls involved there, I don't think this is appropriate. It should quickly be shoved into a queue somewhere, and a reliable (and async) queue processor can handle starting the job.

Reply:
That is what the BackgroundQueue thing is for. Its a "I need to send off this async work and not block my thread" his should use that.

Reply:
We appear to already have a "BackgroundQueue" thing a few lines down. Can we use that?

Reply:
@ChadNedzlek How did it deadlock the Helix API? Based on my understanding, this is essentially just turning the async call into sync.  Anyways, the API itself is async since it eventually devolves into REST API calls.  So at some level it will need to be GetAwaiter().GetResult(), or turned into async here.

Reply:
The issue that I pointed in the first message here is to do exactly that. Use BackgroundQueue as a way to execute `StartAssociatedReleasePipelinesAsync`.

Reply:
The problem with .GetAwaiter().GetResult() is that it blocks the calling thread until the Task that was launched is done. That causes a deadlock if the calling thread is needed to complete the task.

Reply:
Helpful article by our local async expert, Stephen Toub: https://blogs.msdn.microsoft.com/pfxteam/2012/04/13/should-i-expose-synchronous-wrappers-for-asynchronous-methods/

Reply:
oh! An even better one: https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html

TL;DR: Don't ever call GetAwaiter().GetResult().  If you absolutely, super, definitely have to, you need to make sure literally every await in the entire closure of your entire program uses "ConfigureAwait(false)" (which is basically impossible, and you'll probably spend a huge amount of time tracking down random deadlocks as long as the code exists)